Mathlib/Analysis/SpecialFunctions/OrdinaryHypergeometric.lean

```diff
@@ -0,0 +1,221 @@
+  exact ordinaryHypergeometric_radius_top_of_neg_nat₁ 𝔸 a c
+
+theorem ordinaryHypergeometric_radius_top_of_neg_nat₃ {k : ℕ} :
+    (ordinaryHypergeometricSeries 𝔸 a b (-(k : 𝕂))).radius = ⊤ := by
+  refine FormalMultilinearSeries.radius_eq_top_of_forall_image_add_eq_zero _ (1 + k) fun n ↦ ?_
+  exact ordinaryHypergeometricSeries_eq_zero_of_neg_nat a b (-(k : 𝕂)) (by aesop) (by omega)
+
+/-- An iff variation on `ordinaryHypergeometricSeries_eq_zero_of_nonpos_int` for `[RCLike 𝕂]`. -/
+lemma ordinaryHypergeometricSeries_eq_zero_iff (n : ℕ) :
+    ordinaryHypergeometricSeries 𝔸 a b c n = 0 ↔ ∃ k < n, k = -a ∨ k = -b ∨ k = -c := by
+  refine ⟨fun h ↦ ?_, fun zero ↦ ?_⟩
+  · rw [ordinaryHypergeometricSeries, ofScalars_eq_zero] at h
+    simp only [_root_.mul_eq_zero, inv_eq_zero] at h
+    rcases h with ((hn | h) | h) | h
+    · simp [Nat.factorial_ne_zero] at hn
+    all_goals
+      obtain ⟨kn, hkn, hn⟩ := (ascPochhammer_eval_eq_zero_iff _ _).1 h
+      exact ⟨kn, hkn, by tauto⟩
+  · obtain ⟨_, h, hn⟩ := zero
+    exact ordinaryHypergeometricSeries_eq_zero_of_neg_nat a b c hn h
+
+theorem ordinaryHypergeometricSeries_succ_norm_div_norm (n : ℕ)
+    (habc : ∀ kn < n, (↑kn ≠ -a ∧ ↑kn ≠ -b ∧ ↑kn ≠ -c)) :
+      ‖ordinaryHypergeometricCoefficient a b c n.succ‖ / ‖ordinaryHypergeometricCoefficient a b c n‖
+      = ‖a + n‖ * ‖b + n‖ * ‖c + n‖⁻¹ * ‖1 + (n : 𝕂)‖⁻¹ := by
+  simp only [mul_inv_rev, factorial_succ, cast_mul, cast_add,
+    cast_one, ascPochhammer_succ_eval, norm_mul, norm_inv]
+  have : ‖(n ! : 𝕂)‖⁻¹ * ‖(n : 𝕂) + 1‖⁻¹ * (‖Polynomial.eval a (ascPochhammer 𝕂 n)‖ * ‖a + n‖) *
+    (‖Polynomial.eval b (ascPochhammer 𝕂 n)‖ * ‖b + n‖) *
+    (‖c + n‖⁻¹ * ‖Polynomial.eval c (ascPochhammer 𝕂 n)‖⁻¹) /
+    (‖(n ! : 𝕂)‖⁻¹ * ‖Polynomial.eval a (ascPochhammer 𝕂 n)‖ *
+    ‖Polynomial.eval b (ascPochhammer 𝕂 n)‖ * ‖Polynomial.eval c (ascPochhammer 𝕂 n)‖⁻¹) =
+    ‖(n ! : 𝕂)‖⁻¹⁻¹ * ‖(n ! : 𝕂)‖⁻¹ * ‖Polynomial.eval a (ascPochhammer 𝕂 n)‖ *
+    ‖Polynomial.eval a (ascPochhammer 𝕂 n)‖⁻¹ * ‖Polynomial.eval b (ascPochhammer 𝕂 n)‖ *
+    ‖Polynomial.eval b (ascPochhammer 𝕂 n)‖⁻¹ * ‖Polynomial.eval c (ascPochhammer 𝕂 n)‖⁻¹⁻¹ *
+    ‖Polynomial.eval c (ascPochhammer 𝕂 n)‖⁻¹ * ‖a + n‖ * ‖b + n‖ * ‖c + n‖⁻¹ * ‖1 + (n : 𝕂)‖⁻¹ :=
+    by ring_nf
+  rw [this, inv_inv, inv_inv]
+  repeat rw [DivisionRing.mul_inv_cancel, one_mul]
+  all_goals
+    rw [norm_ne_zero_iff]
+  any_goals
+    apply (ascPochhammer_eval_eq_zero_iff n _).not.2
+    push_neg
+    exact fun kn hkn ↦ by simp [habc kn hkn]
+  exact cast_ne_zero.2 (factorial_ne_zero n)
+
+/-- The ratio of successive terms of `ordinaryHypergeometricSeries` tends to one. This theorem
+is used in the proof `ordinaryHypergeometric_ratio_tendsto_nhds_atTop`. -/
+theorem ordinaryHypergeometricSeries_ratio_tendsto_one_atTop :
+    Tendsto (fun k : ℕ ↦ (a + k) * (b + k) * (c + k)⁻¹ * ((1 : 𝕂) + k)⁻¹) atTop (𝓝 1) := by
+  conv =>
+    enter [1, n]
+    rw [mul_assoc, ← mul_inv, ← div_eq_mul_inv, mul_div_mul_comm]
```

Comment:
There's no need to work this hard (using `conv` and manual rewriting):
```suggestion
  have (k : ℕ) : (a + k) * (b + k) * (c + k)⁻¹ * ((1 : 𝕂) + k)⁻¹ =
      (a + k) / (c + k) * ((b + k) / (1 + k)) := by field_simp
  simp_rw [this]
```
I also think it's easier to read this.